articles/113_ros_time.md

```diff
@@ -0,0 +1,154 @@
+
+##### No Advanced Estimating Clock
+
+There are more advanced techniques which could be included to attempt to estimate the propagation properties and extrapolate between time ticks.
+However all of these techniques will require making assumptions about the future behavior of the time abstraction.
+And in the case that playback or simulation is instantaneously paused, it will break any of these assumptions.
+There are techniques which would allow potential interpolation, however to make these possible it would require providing guarantees about the continuity of time into the future.
+For more accuracy the progress of time can be slowed, or the frequency of publishing can be increased.
+Tuning the parameters for the `/clock` topic lets you trade off time for computational effort and/or bandwidth.
+
+#### Custom Time Source
+
+It is possible that the user may have access to an out of band time source which can provide better performance than the default source the `/clock` topic.
+A potential alternative source of time might be a GPS.
+The user will be able to switch out the time source for either each instance of their Time object as well as the ability to override the default for the process.
+
+## Implementation
+
+The public API will be provided by the client libraries.
+The client libraries will be able to rely on a base implementation in rcl which provides a common behavior.
+If the client library implementer chooses they my also reimplement the core behavior, however it is expected that the rcl implementation should be adaquate to support any language with a foreign function interface for C code.
+
+### Public API
+
+Each client library is expected to provide parallel interfaces for each of the three abstract types of time.
+It is up to the implementation to choose the most relevant appropriate way to select the parallel implementations.
+
+In each client library will provide `Time`, `Duration`, and `Rate` implementations.
+The duration abstraction will support a `sleep_for` function as well as a `sleep_until` method.
+It will also provide a `Timer` object which will provide periodic callback functionality using the ROS time abstraction.
+
+### RCL implementation
+
+In `rcl` there will be representatative datatypes and methods to implement each of the three time abstractions for each of the core datatypes.
+However at the `rcl` level the time API will not have a threading model and will rely on the client library implementation to do any threading such as is required by sleep methods.
+
+It will provide implementations parallel to the public datastructures and storage which the client library can depend upon and to which it can delegate.
+The underlying datatypes will also provide ways to register notifications, however it is the responsibility of the client library implementation to collect and dispatch user callbacks.
```

Comment:
For the previous three sentences I understand what you're saying, but I think others will find it harder to follow. Consider either removing these technical details of the implementation or describe them more thoroughly, perhaps with examples.